### PR TITLE
Brave Leo button on the toolbar toggles the sidebar AI tab.

### PR DESCRIPTION
**Pull request by boocmp**
Resolves https://github.com/brave/brave-browser/issues/32610

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

File: browser/ui/webui/speedreader/speedreader_toolbar_data_handler_impl.cc [hunk 1]
```diff
@@ -137,7 +137,7 @@ void SpeedreaderToolbarDataHandlerImpl::AiChat() {
     return;
   }
 #if BUILDFLAG(ENABLE_AI_CHAT)
-  static_cast<BraveBrowserWindow*>(browser_->window())->OpenAiChatPanel();
+  static_cast<BraveBrowserWindow*>(browser_->window())->ToggleAiChatPanel();
```
Comment (simonhong) [inline browser/ui/webui/speedreader/speedreader_toolbar_data_handler_impl.cc:140]: I think we don't need to plumbing to view layer.
We can use `SidePanelUI::GetSidePanelUIForBrowser()->Show()` and `->GetCurrentEntryId()` for toggling.

Comment (simonhong) [reply browser/ui/webui/speedreader/speedreader_toolbar_data_handler_impl.cc:140]: Ah.. maybe this was implemented initially when `SidePanelUI` was in view layer.

Comment (boocmp) [reply browser/ui/webui/speedreader/speedreader_toolbar_data_handler_impl.cc:140]: Thanks, I'll remove the view layer

**Review by simonhong — APPROVED** [review]

File: components/resources/speedreader_strings.grdp [hunk 2]
```diff
@@ -38,7 +38,7 @@
       <message name="IDS_READER_MODE_TEXT_TO_SPEECH_FORWARD"  desc="Tooltip for forward button">Forward</message>
       <message name="IDS_READER_MODE_TEXT_TO_SPEECH_SPEED_DECREASE"  desc="Tooltip for decrease playback speed button">Decrease the playback speed</message>
       <message name="IDS_READER_MODE_TEXT_TO_SPEECH_SPEED_INCREASE"  desc="Tooltip for increase playback speed button">Increase the playback speed</message>
-      <message name="IDS_READER_MODE_AI" desc="Tooltop for AI button">Brave AI</message>
+      <message name="IDS_READER_MODE_AI" desc="Tooltop for AI button">Brave Leo</message>
```
Comment (mkarolin) [inline components/resources/speedreader_strings.grdp:41]: Toolt**i**p

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**